tests/functional/directives.ts

```diff
@@ -0,0 +1,655 @@
+// tslint:disable:member-ordering
+import "reflect-metadata";
+
+import {
+  GraphQLSchema,
+  FieldDefinitionNode,
+  GraphQLField,
+  GraphQLDirective,
+  DirectiveLocation,
+  GraphQLString,
+  parseValue,
+  graphql,
+  GraphQLInputObjectType,
+  InputValueDefinitionNode,
+  InputObjectTypeDefinitionNode,
+  GraphQLInputField,
+  GraphQLScalarType,
+  GraphQLNonNull,
+} from "graphql";
+import {
+  Field,
+  InputType,
+  Resolver,
+  Query,
+  Arg,
+  Directive,
+  buildSchema,
+  ObjectType,
+  Mutation,
+} from "../../src";
+import { getMetadataStorage } from "../../src/metadata/getMetadataStorage";
+import Maybe from "graphql/tsutils/Maybe";
+import { SchemaDirectiveVisitor } from "graphql-tools";
+import { ApolloServer } from "apollo-server";
+
+class UpperCaseType extends GraphQLScalarType {
```

Comment:
I though it's recommended to use create custom scalars this way:
https://github.com/19majkel94/type-graphql/blob/0103091f903f9a38e7d94aed1be9ecf51923b6ca/src/scalars/isodate.ts#L3-L4

Reply:
I got most of my code from the `apollo` directives section on their website.